bot/monkey_patches.py

```diff
@@ -50,3 +52,36 @@ async def honeybadger_type(self, channel_id: int) -> None:  # noqa: ANN001
             pass
 
     http.HTTPClient.send_typing = honeybadger_type
+
+
+def patch_message_converter() -> None:
+    """
+    Make the Message converter infer channelID from the given context.
+
+    Discord.py's Message converter is supposed to infer channelID based
+    on ctx.channel if only a messageID is given. A "refactor" (linked below)
+    a few weeks before d.py's archival broke this, so that if only a messageID
+    is given to the converter, it will only find that message if it's in the bot's
+    cache.
+
+    https://github.com/Rapptz/discord.py/commit/1a4e73d59932cdbe7bf2c281f25e32529fc7ae1f
```

Comment:
I don't exactly see what line is patching this, given I can't find an await to fetch messages from the provided channel. If they aren't in the cache, how would not fetching help? Serious question, because I can't exactly see where this solves the problem.

Reply:
This patches the Message converter, not fetching a message from the API. This is the doc string for Discord.py's message converter https://github.com/Rapptz/discord.py/blob/master/discord/ext/commands/converter.py#L373-L386.

The second step there doesn't actually work. It doesn't fall back to the context channel if just a messageID is given, it actually only checks the cache.

Line 84 of this PR is what is actually patching this behaviour, where if _get_as_snowflake doesn't return a channel, we use the channel from the context instead, so that the message converter actually works as intended.